pkg/cloud-provider/cloudapi/aws/aws_security.go

```diff
@@ -214,36 +216,31 @@ func (ec2Cfg *ec2ServiceConfig) realizeIngressIPPermissions(cloudSgObj *ec2.Secu
 		newIpPermissions = append(newIpPermissions, ipPermission)
 	}
 
-	ipPermissionsToAdd, ipPermissionsToDelete := computeIPPermissionDelta(newIpPermissions, cloudSgObj.IpPermissions)
-	awsPluginLogger().V(1).Info("ingress", "add", ipPermissionsToAdd, "delete", ipPermissionsToDelete)
-
-	// revoke old ingress rules and add new rules
-	if len(ipPermissionsToDelete) > 0 {
-		request := &ec2.RevokeSecurityGroupIngressInput{
-			GroupId:       cloudSgObj.GroupId,
-			IpPermissions: ipPermissionsToDelete,
-		}
-		if _, err := ec2Cfg.apiClient.revokeSecurityGroupIngress(request); err != nil {
+	if len(newIpPermissions) > 0 {
+		if isDelete {
+			request := &ec2.RevokeSecurityGroupIngressInput{
+				GroupId:       cloudSgObj.GroupId,
+				IpPermissions: newIpPermissions,
+			}
+			_, err := ec2Cfg.apiClient.revokeSecurityGroupIngress(request)
 			return err
-		}
-	}
-
-	if len(ipPermissionsToAdd) > 0 {
-		request := &ec2.AuthorizeSecurityGroupIngressInput{
-			GroupId:       cloudSgObj.GroupId,
-			IpPermissions: ipPermissionsToAdd,
-		}
-		if _, err := ec2Cfg.apiClient.authorizeSecurityGroupIngress(request); err != nil {
+		} else {
+			request := &ec2.AuthorizeSecurityGroupIngressInput{
+				GroupId:       cloudSgObj.GroupId,
+				IpPermissions: newIpPermissions,
+			}
+			_, err := ec2Cfg.apiClient.authorizeSecurityGroupIngress(request)
 			return err
 		}
 	}
 	return nil
 }
 
-func (ec2Cfg *ec2ServiceConfig) realizeEgressIPPermissions(cloudSgObj *ec2.SecurityGroup, rules []*securitygroup.EgressRule,
-	cloudSGNameToObj map[string]*ec2.SecurityGroup) error {
+func (ec2Cfg *ec2ServiceConfig) realizeEgressIPPermissions(cloudSgObj *ec2.SecurityGroup, rules []*securitygroup.CloudRule,
```

Comment:
Scan and add comments on functions, as its missed earlier

Reply:
All functions in cloud plugin don't have comments in front. Do you want to add comments to all of them now?

Reply:
keep doing for functions as and when you update it and you find it missing

Reply:
Done for all functions I touched